src/rajaperf_config.hpp.in

```diff
@@ -27,41 +27,24 @@
 namespace rajaperf {
 
 struct configuration {
-
-// Version of RAJA Perf Suite (ex: 0.1.0)
-static const std::string perfsuite_version =
-"@RAJA_PERFSUITE_VERSION_MAJOR@" + std::string(".") +
-"@RAJA_PERFSUITE_VERSION_MINOR@" + std::string(".") +
-"@RAJA_PERFSUITE_VERSION_PATCHLEVEL@";
-
-// Version of RAJA used to build (ex: 0.2.4)
-static const std::string raja_version = 
-std::to_string(RAJA::RAJA_VERSION_MAJOR) + std::string(".") +
-std::to_string(RAJA::RAJA_VERSION_MINOR) + std::string(".") +
-std::to_string(RAJA::RAJA_VERSION_PATCH_LEVEL);
-
-// Systype and machine code was built on (ex: chaos_5_x64_64, rzhasgpu18)
-static const std::string systype_build = "@RAJAPERF_BUILD_SYSTYPE@";
-static const std::string machine_build = "@RAJAPERF_BUILD_HOST@";
-		
-// Compiler used to build (ex: gcc-4.9.3)
-static const std::string compiler = "@RAJAPERF_COMPILER@";
-
-// Command options used to build (ex: -Ofast -mavx)
-static const std::string compiler_options = "@RAJAPERF_COMPILER_OPTIONS@";
-		
-// Name of user who ran code
-std::string user_run;
-
-// Date, time code was run
-std::string date_run;
-
-// Systype and machine code ran on (ex: chaos_5_x64_64)
-std::string systype_run;
-std::string machine_run;
-		
+constexpr static const char* perfsuite_version = "@CMAKE_PROJECT_VERSION@";
```

Comment:
These were changed to support the ability to read in the "configured" rajaperf_config.hpp for use in RAJAPerf itself during compilation, vs just a file you could look as part of a run set. I'll take another look to be sure. You would need c++17 to do something similar: constexpr std::string_view sv = "hello, world".  But, depending on compiler constexpr std:string may generate: 

    error: the type ‘const string {aka const std::basic_string}’ of constexpr variable ‘constString’ is not literal... because... ‘std::basic_string’ has a non-trivial destructor

std::string will become constexpr in C++20